app/src/blocks/blocks.service.ts

```diff
@@ -10,6 +10,7 @@ import { ApiError } from "src/utils/ApiError";
 
 @Injectable()
 export class BlocksService {
+  readonly logger = new Logger(BlocksService.name);
```

Comment:
Is there not a way to inject this into the service via the constructor?

Reply:
I took a little bit of time to read into the docs.  What David is doing here seems to be the most simple approach suggested by their docs.  It's quite impressive how non-opinionated NestJS is.

Looks like we can use the Logger class as David is doing here wherever we'd like -- but there is not defined/consistent pattern.  Seems like a logger comes shipped out of the box using a singleton pattern or we can inject a new instance in each module/service.  Dep injection seems to require us to create a custom logger class with the `@injectable` decorator.

I think we should future proof with a custom logger that's injected.  Thoughts @mgibson-casper ?

Reply:
Well we are applying the `Injectable` here and in all other service classes using the logger. I also just took a quick look and this seems OK to me so I'm good to keep as is 👍